packages/camera/camera/lib/src/camera_controller.dart

```diff
@@ -468,6 +476,16 @@ class CameraController extends ValueNotifier<CameraValue> {
     }
   }
 
+  /// Sets the flash mode for taking pictures.
+  Future<void> setFlashMode(FlashMode mode) async {
+    try {
+      await CameraPlatform.instance.setFlashMode(_cameraId, mode);
+      value = value.copyWith(flashMode: mode);
+    } on PlatformException catch (e) {
+      throw CameraException(e.code, e.message);
+    }
+  }
```

Comment:
We should add unit tests for this method:

1. One to test if the method correctly catches the `PlatformException` and converts it to an instance of the `CameraException` class;
2. One to test if the method correctly calls the `setFlashMode` method on the `CameraPlatform` instance.